cmd/git/main.go

```diff
@@ -316,6 +318,36 @@ func clone(ctx context.Context) error {
 		}
 	}
 
+	if flagValues.gitURLRewrite {
+		var hostname string
+		switch {
+		case strings.HasPrefix(flagValues.url, "git@"):
+			trimmed := strings.TrimPrefix(flagValues.url, "git@")
+			splitted := strings.SplitN(trimmed, ":", 2)
+			hostname = splitted[0]
+
+		case strings.HasPrefix(flagValues.url, "http"):
+			repoURL, err := url.Parse(flagValues.url)
+			if err != nil {
+				return err
+			}
+			hostname = repoURL.Host
+
+		default:
+			return fmt.Errorf("unknown URL type: %q", flagValues.url)
```

Comment:
Not sure if this is a good idea. If you check https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats, then `ssh://` and `git://` (I think GitHub just stopped supporting it) would also be possible. We could add more cases. Anyway, imo we should just print a warning that we cannot parse it and therefore do not configure a rewrite.

Reply:
GitHub did in fact stop supporting (introducing "brown outs") for `git://`

Reply:
Changed implementation to print a message in case it is an unknown or unsupported type.